fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -364,7 +364,7 @@ private static Type fromProtoType(@Nullable Descriptors.GenericDescriptor descri
             return fromProtoTypeToArray(descriptor, protoType, typeCode, false);
         } else {
             if (typeCode.isPrimitive()) {
-                return primitiveType(typeCode, isNullable);
+                return primitiveType(typeCode, false);
```

Comment:
at [Line#2081](https://github.com/FoundationDB/fdb-record-layer/blob/main/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java#L2081), `fromProtoType(...., isNullable = !fieldDescriptor.isRequired()`, conflicts with other places where primitive types are NOT nullable (isNullable = false), so I changed the logic here to make all primitiveTypes not nullable.